rasa/shared/core/slots.py

```diff
@@ -160,6 +160,21 @@ def feature_dimensionality(self) -> int:
         return len(self.as_feature())
 
 
+def bool_from_any(x: Any) -> bool:
+    """ Converts bool/float/int/str to bool or raises TypeError """
+    
+    if isinstance(x, bool):
+        return x
+    elif isinstance(x, (float, int)):
+        return x == 1.
+    elif isinstance(x, str) and x.isnumeric():
+        return float(x) == 1.
+    elif isinstance(x, str):
+        return x.strip().lower() == "true"
```

Comment:
this will happily accept slot values like `"Berlin"` and set the boolean slot to `0.0`.

I think we should explicitly test for both cases, `"true"` and `"false"` strings and raise `ValueError` otherwise